aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsCustomizableOperationDecorator.kt

```diff
@@ -0,0 +1,101 @@
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+
+class CustomizableOperationTestHelpers(runtimeConfig: RuntimeConfig) :
+    CustomizableOperationCustomization() {
+    private val codegenScope = arrayOf(
+        *RuntimeType.preludeScope,
+        "AwsUserAgent" to AwsRuntimeType.awsHttp(runtimeConfig)
+            .resolve("user_agent::AwsUserAgent"),
+        "BeforeTransmit" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::context::phase::BeforeTransmit"),
+        "ConfigBag" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("config_bag::ConfigBag"),
+        "ConfigBagAccessors" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("client::orchestrator::ConfigBagAccessors"),
+        "InterceptorContext" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::InterceptorContext"),
+        "RequestTime" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("client::orchestrator::RequestTime"),
+        "SharedInterceptor" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::SharedInterceptor"),
+        "TestParamsSetterInterceptor" to CargoDependency.smithyRuntime(runtimeConfig).withFeature("test-util")
+            .toType().resolve("client::test_util::interceptor::TestParamsSetterInterceptor"),
+    )
+
+    override fun section(section: CustomizableOperationSection): Writable =
+        writable {
+            if (section is CustomizableOperationSection.CustomizableOperationImpl) {
+                if (section.operationShape == null) {
+                    // This branch customizes CustomizableOperation in the middleware. section.operationShape being
+                    // null means that this customization is rendered in a place where we don't need to figure out
+                    // the module for an operation (which is the case for CustomizableOperation in the middleware
+                    // that is rendered in the customize module).
+                    rust(
+                        """
+                        ##[doc(hidden)]
+                        // This is a temporary method for testing. NEVER use it in production
+                        pub fn request_time_for_tests(mut self, request_time: ::std::time::SystemTime) -> Self {
+                            self.operation.properties_mut().insert(request_time);
+                            self
+                        }
+
+                        ##[doc(hidden)]
+                        // This is a temporary method for testing. NEVER use it in production
+                        pub fn user_agent_for_tests(mut self) -> Self {
+                            self.operation.properties_mut().insert(aws_http::user_agent::AwsUserAgent::for_tests());
```

Comment:
`AwsUserAgent` needs to be referred to via `RuntimeType`

Reply:
Addressed in 3074cdb.